### PR TITLE
Added Bitraverse for Ior (#2144)

### PR DESCRIPTION
**Pull request by V-Lamp**
Added `Bitraverse[Ior]`, as discussed in #2144 

For the testing part, I followed the pattern found in for testing `Bitraverse[Either]`, within `cats.tests.EitherSuite`

This is my first PR, let me know if I need do something around this.
...and thanks for all the good work!

### CODE REVIEW

File: core/src/main/scala/cats/data/Ior.scala [hunk 1]
```diff
@@ -145,6 +145,23 @@ object Ior extends IorInstances with IorFunctions {
 }
 
 private[data] sealed abstract class IorInstances extends IorInstances0 {
+
+  implicit val catsBitraverseForIor: Bitraverse[Ior] = new Bitraverse[Ior] {
+
+    def bitraverse[G[_], A, B, C, D](fab: Ior[A, B])(f: A => G[C], g: B => G[D])(implicit G: Applicative[G]): G[Ior[C, D]] =
+      fab.fold(
```
Comment (LukaJCB) [inline core/src/main/scala/cats/data/Ior.scala]: Can we replace this `fold` with an explicit pattern match?

File: core/src/main/scala/cats/data/Ior.scala [hunk 2]
```diff
@@ -145,6 +145,23 @@ object Ior extends IorInstances with IorFunctions {
...
+        a => G.map(f(a))(Ior.Left(_)),
+        b => G.map(g(b))(Ior.Right(_)),
+        (a, b) => G.map2(f(a), g(b))(Ior.Both(_, _))
+      )
+
+    def bifoldLeft[A, B, C](fab: Ior[A, B], c: C)(f: (C, A) => C, g: (C, B) => C): C =
+      fab.fold(f(c, _), g(c, _), (a, b) => g(f(c, a), b))
```
Comment (LukaJCB) [inline core/src/main/scala/cats/data/Ior.scala]: Same here.

File: core/src/main/scala/cats/data/Ior.scala [hunk 3]
```diff
@@ -145,6 +145,23 @@ object Ior extends IorInstances with IorFunctions {
...
+
+    def bifoldRight[A, B, C](fab: Ior[A, B], c: Eval[C])(f: (A, Eval[C]) => Eval[C], g: (B, Eval[C]) => Eval[C]): Eval[C] =
+      fab.fold(f(_, c), g(_, c), (a, b) => g(b, f(a, c)))
```
Comment (LukaJCB) [inline core/src/main/scala/cats/data/Ior.scala]: and here.

**Comment by V-Lamp** [comment]
Thanks @LukaJCB, done

**Pull request merged**